src/Features/Core/Portable/InlineMethod/AbstractInlineMethodRefactoringProvider.cs

```diff
@@ -0,0 +1,192 @@
+
+        private async Task<SyntaxNode?> GetInvocationExpressionSyntaxNodeAsync(CodeRefactoringContext context)
+        {
+            var syntaxNode = await context.TryGetRelevantNodeAsync<TInvocationSyntaxNode>().ConfigureAwait(false);
+            return syntaxNode;
+        }
+
+        private static bool IsExpressionSyntax(SyntaxNode syntaxNode)
+            => syntaxNode is TExpressionSyntax;
+
+        private async Task<Document> InlineMethodAsync(
+            Document document,
+            SyntaxNode calleeMethodInvocationSyntaxNode,
+            IMethodSymbol calleeMethodSymbol,
+            SyntaxNode calleeMethodDeclarationSyntaxNode,
+            CancellationToken cancellationToken)
+        {
+            var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            var methodParametersInfo = MethodParametersInfo.GetMethodParametersInfo(
+                this,
+                _syntaxFacts,
+                semanticModel,
+                calleeMethodInvocationSyntaxNode,
+                calleeMethodSymbol,
+                cancellationToken);
+
+            var root = await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var inlineContext = await InlineMethodContext.GetInlineContextAsync(
+                this,
+                _syntaxFacts,
+                _precedenceService,
+                document,
+                semanticModel,
+                calleeMethodInvocationSyntaxNode,
+                calleeMethodSymbol,
+                calleeMethodDeclarationSyntaxNode,
+                methodParametersInfo,
+                root,
+                cancellationToken).ConfigureAwait(false);
+
+            var documentEditor = await DocumentEditor.CreateAsync(document, cancellationToken).ConfigureAwait(false);
+            var statementContainsCalleeInvocationExpression = inlineContext.StatementContainingCallee;
+            // In case can't find the containing statement, don't do insertion
+            if (statementContainsCalleeInvocationExpression != null)
+            {
+                foreach (var statement in inlineContext.StatementsToInsertBeforeCallee)
+                {
+                    documentEditor.InsertBefore(
+                        statementContainsCalleeInvocationExpression,
+                        // Make sure the statement is aligned with the existing statement
+                        statement.WithLeadingTrivia(statementContainsCalleeInvocationExpression.GetLeadingTrivia()));
+                }
+            }
+
+            var syntaxNodeToReplace = inlineContext.SyntaxNodeToReplace;
+            var inlineSyntaxNode = inlineContext.InlineSyntaxNode;
+            if (syntaxNodeToReplace != null)
+            {
+                if (inlineSyntaxNode == null)
+                {
+                    // When it has only one return statement in the callee & return void, just remove the whole statement.
```

Comment:
example/.

Reply:
Filter out this case during the check as we dicussed some other comments:)